data_source.go

```diff
@@ -193,9 +212,101 @@ FROM
 		nodeInfo.Xlog.ReceivedLocation = nodeInfo.Xlog.ReplayedLocation
 	}
 
+	pgCurrentWalLsn, err := ds.getPgCurrentWalLsn(nodeInfo.Role)
+	if err != nil {
+		log.Fatalln("Error getting pg_current_wal_lsn:", err)
+	}
+
+	pgLastWalLsn, err := ds.getPgLastWalReceiveLsn()
+	if err != nil {
+		log.Fatalln("Error getting pg_last_wal_lsn:", err)
+	}
+
+	byteLag, err := ds.getPgWalLsnDiff(pgCurrentWalLsn, pgLastWalLsn)
+	if err != nil {
+		log.Fatalln("Error getting pg_wal_lsn_diff:", err)
+	}
+
+	nodeInfo.ByteLag = byteLag
+
 	return nodeInfo, nil
 }
 
+func (ds *pgDataSource) getConnInfo() (string, error) {
```

Comment:
Should we call this `getUpstreamConnInfo` or `getWalSenderConnInfo`? Something to indicate this conninfo is not ours, but the server sending us data.

Reply:
Great point. Renamed to `getUpstreamConnInfo`.